block/initchain.go

```diff
@@ -3,13 +3,17 @@ package block
 import (
 	"context"
 	"errors"
+	"fmt"
 
 	tmtypes "github.com/tendermint/tendermint/types"
 )
 
 func (m *Manager) RunInitChain(ctx context.Context) error {
-	// get the proposer's consensus pubkey
-	proposer := m.SLClient.GetProposer()
+	// Get the proposer at the initial height
+	proposer, err := m.SLClient.GetProposerAtHeight(1)
```

Comment:
from what i understand, this new function will get the proposer from the state info. in case there is no state info i guess it will return not found, and therefore it will not be possible to run initchain for the sequencer, or other nodes when there  is no state update yet. i guess we need to fallback to default getproposer() when there is no state update yet.

Reply:
I think I'll follow the array logic here where `0` will be the initial proposer and `-1` will be the last. 
we can also have 3 seperate methods (i.e initial, at height, last) but I don't think it's crucuial for now and it's more methods to add to the interface.